assets/src/apps/delivery/components/ActivityRenderer.tsx

```diff
@@ -327,12 +362,48 @@ const ActivityRenderer: React.FC<ActivityRendererProps> = ({
   const currentActivityId = useSelector(selectCurrentActivityId);
   const initPhaseComplete = useSelector(selectInitPhaseComplete);
   const initStateFacts = useSelector(selectInitStateFacts);
+
+  const updateGlobalState = async (snapshot: any, stateFacts: any) => {
+    const payloadData = {} as any;
+    stateFacts.map((fact: string) => {
+      // EverApp Information
+      if (fact.startsWith('app.')) {
+        const data = fact.split('.');
+        const objId = data.splice(2).join('.');
+        const value = snapshot[fact];
+        payloadData[data[1]] = { ...payloadData[data[1]], [objId]: value };
+      }
+    });
+
+    const arrayPLData = Object.keys(payloadData);
+    const retrievedData = (await Extrinsic.readGlobal(arrayPLData)) as any;
+    const parsedRetrievedData = JSON.parse(retrievedData);
```

Comment:
this will throw an exception when there is no data; also we should use the same `updateGlobalUserState` here which is designed to do the reading first

Reply:
Addressed.